File: src/main/java/sample/model/search/RuleSearchParameters.java
package sample.model.search;

import sample.model.rules.RuleFamily;
import sample.model.simulation.Grid;

public class RuleSearchParameters extends SearchParameters {
    private final RuleFamily minRule;
    private final RuleFamily maxRule;
    private final int maxPeriod;
    private final Grid targetPattern;

    public RuleSearchParameters(Grid pattern, RuleFamily minRule, RuleFamily maxRule, int maxPeriod) {
        this.minRule = minRule;
        this.maxRule = maxRule;
        this.targetPattern = pattern;
        this.maxPeriod = maxPeriod;
    }

    // Accessors
    public Grid getTargetPattern() {
        return targetPattern;
    }

    public int getMaxPeriod() {
        return maxPeriod;
    }

    public RuleFamily getMinRule() {
        return minRule;
    }

    public RuleFamily getMaxRule() {
        return maxRule;
    }
}

File: src/main/kotlin/application/model/search/rulesrc/RuleSearchParameters.kt
package application.model.search.rulesrc

import application.model.search.rulesrc.RuleSearchParameters
import application.model.search.SearchProgram
import java.util.HashSet
import kotlin.Throws
import java.lang.IllegalArgumentException
import application.model.simulation.Simulator
import application.model.rules.MinMaxRuleable
import application.model.Coordinate
import application.model.patterns.Spaceship
import java.io.FileWriter
import application.model.rules.RuleFamily
import application.model.database.GliderDBEntry
import application.model.patterns.Oscillator
import java.io.IOException
import java.util.logging.LogManager
import application.model.search.SearchParameters
import application.model.simulation.Grid

class RuleSearchParameters(// Accessors
    val targetPattern: Grid, val minRule: RuleFamily, val maxRule: RuleFamily, val maxPeriod: Int,
    val minPop: Int, val maxPop: Int, val maxX: Int, val maxY: Int
) : SearchParameters()